mattermost/docker-compose.yml

```
@@ -0,0 +1,43 @@
+version: "3.7"
+
+services:
+  app_proxy:
+    environment:
+      APP_HOST: mattermost_app_1
+      APP_PORT: 8000
+      PROXY_AUTH_ADD: "false"
+  
+  app:
+    image: alexlack/mattermost-app:v10.1.2@sha256:fa306060de89dce0d619537bc8cecc63289008f6ad719d65d5b83404ee8831d0
+    user: "1000:1000"
```

It looks looks like the `mattermost-app` container must run as root or else the app will fail to start:

```
$ sudo docker logs -f mattermost_app_1
Fixing permissions on /mattermost/data
Fixing permissions on /mattermost/logs
Fixing permissions on /mattermost/config
Fixing permissions on /mattermost/plugins
Fixing permissions on /mattermost/client/plugins
su: must be suid to work properly
```

While not ideal, let's run as root and leave a comment:
```suggestion
    # Needs to run as root or the container fails to run
    # user: "1000:1000"
```

---

Did it fail to start for you? I had it running with no issues. Also have it running as non-root via podman on a different raspberry pi of mine.

---

Yep, it fails to start with `su: must be suid to work properly`.  The app container gets stuck in a reboot loop.

> I had it running with no issues. Also have it running as non-root via podman on a different raspberry pi of mine.

For your podman install, were you using your `alexlack/mattermost-app` image and `user: "1000:1000"` in the compose file? Your initial submission before linting changes didn't have `user: "1000:1000"` https://github.com/getumbrel/umbrel-apps/pull/1714/commits/21fa8503e35bad8cdc599f12fca19d79c2f468c7.
Could you please give it a quick fresh install check to see if you get the same error as me?

---

It's this entrypoint file that it is erroring on when trying to run as non-root: https://github.com/al-lac/mattermost-docker/blob/main/app/entrypoint.sh

---

Running as non-root isn't a dealbreaker right now, so if you don't want to dig into the image too much, then we can ship this without `user: "1000: 1000"`

---

True, also failed for me. Really missed to test it after changing the user variable. Sorry about that!

Made some changes so it works also when user is set, see here: https://github.com/al-lac/mattermost-docker/blob/main/app/priv-entrypoint.sh

Published a new build and updated the sha256 in the PR.

Worked nicely on my pi right now.

---

Excellent, nice work on the fix!

I have tested on arm and x86 and it is working well.